File: src/main/java/org/jdc/template/model/database/main/individual/IndividualManager.java
/*
 * IndividualManager.java
 *
 * Generated on: 10/02/2012 01:40:29
 *
 */



package org.jdc.template.model.database.main.individual;


import org.jdc.template.model.database.DatabaseManager;
import org.threeten.bp.LocalDateTime;
import javax.annotation.Nonnull;
import javax.annotation.Nullable;
import javax.inject.Inject;

@javax.inject.Singleton
public class IndividualManager extends IndividualBaseManager {

    @Inject
    public IndividualManager(DatabaseManager databaseManager) {
        super(databaseManager);
    }

    @Override
    public boolean save(@Nonnull String databaseName, @Nullable Individual individual) {
        if (individual == null) {
            return false;
        }

        if (individual.getBirthDate() != null) {
            individual.setSampleDateTime(LocalDateTime.of(individual.getBirthDate(),  individual.getAlarmTime()));
        }

        individual.setLastModified(LocalDateTime.now());
        return super.save(databaseName, individual);
    }
}

File: src/main/kotlin/org/jdc/template/model/database/main/individual/IndividualManager.kt
/*
 * IndividualManager.kt
 *
 * Generated on: 02/14/2016 07:02:10
 *
 */



package org.jdc.template.model.database.main.individual

import org.jdc.template.model.database.DatabaseManager
import org.threeten.bp.LocalDateTime
import javax.inject.Inject


@javax.inject.Singleton
class IndividualManager @Inject constructor(databaseManager: DatabaseManager) : IndividualBaseManager(databaseManager) {


    override fun save(individual: Individual?, databaseName: String): Boolean {
        if (individual == null) {
            return false
        }

        if (individual.birthDate != null && individual.alarmTime != null) {
            individual.sampleDateTime = LocalDateTime.of(individual.birthDate, individual.alarmTime)
        }

        individual.lastModified = LocalDateTime.now()
        return super.save(individual, databaseName)
    }
}